Geometry/HcalTowerAlgo/plugins/HcalTrigTowerGeometryESProducer.cc

```diff
@@ -23,6 +23,10 @@ HcalTrigTowerGeometryESProducer::produce( const CaloGeometryRecord & iRecord )
     bool enableRCTHF=(tmode==HcalTopologyMode::TriggerMode_2009 || tmode==HcalTopologyMode::TriggerMode_2016);
     bool enable1x1HF=(tmode==HcalTopologyMode::TriggerMode_2016 || tmode==HcalTopologyMode::TriggerMode_2017);
     m_hcalTrigTowerGeom->setupHFTowers(enableRCTHF,enable1x1HF);
+    // TODO this piggy-backs on the HF versioning, which corresponds to
+    // 2017 TP --- trigger modes should really be numbered by year.
+    bool enable2017HE = enable1x1HF;
+    m_hcalTrigTowerGeom->setupHETowers(enable2017HE);
```

Comment:
@matz-e - this public API is kind of dangerous. An object variable is set based on a topology mode in an object constructed from the topology. The topology will not change, however the API allows it to change effectively allowing perceptive change of the topology. Other objects will see different topology depending on if they retrieve it from the geometry or from the geometry's topology.